v2/tools/generator/internal/jsonast/swagger_type_extractor.go

```diff
@@ -191,7 +191,10 @@ func (extractor *SwaggerTypeExtractor) extractOneResourceType(
 	var resourceSpec astmodel.Type
 	if specSchema == nil {
 		// nil indicates empty body
-		resourceSpec = astmodel.NewObjectType()
+		// Fabricate a TypeName here because that's what all other specs are (typename pointing to parameters type)
+		name := astmodel.MakeInternalTypeName(resourceName.InternalPackageReference(), fmt.Sprintf("%sCreateParameters", resourceName.Name()))
```

Comment:
Should this use an IdFactory instead?

Reply:
I don't think it's required, as we're really just appending a suffix here to an already generated name. This is similar to how we append names for `StatusSuffix`, which does:
`resourceName.WithName(resourceName.Name() + astmodel.StatusSuffix)`